providers/implementations/signature/rsa.c

```diff
@@ -346,8 +375,9 @@ static int rsa_signverify_init(void *vprsactx, void *vrsa, int operation)
                 }
                 prsactx->saltlen = min_saltlen;
 
-                return rsa_setup_md(prsactx, mdname, prsactx->propq)
-                    && rsa_setup_mgf1_md(prsactx, mgf1mdname, prsactx->propq)
+                return rsa_setup_mgf1_md(prsactx, mgf1mdname, prsactx->propq)
+                    /* must be called before rsa_setup_md for PSS */
+                    && rsa_setup_md(prsactx, mdname, prsactx->propq)
```

Comment:
The way I read lines of code, a comment on its own line relates to what follows, so that comment is telling me that rsa_setup_md must be called before rsa_setup_md...